isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -4350,6 +4406,9 @@ namespace isobus
 
 	void VirtualTerminalClient::process_command_queue()
 	{
+#if !defined CAN_STACK_DISABLE_THREADS && !defined ARDUINO
+		std::lock_guard<std::mutex> lock(commandQueueMutex);
+#endif
 		for (auto it = commandQueue.begin(); it != commandQueue.end();)
 		{
 			if (send_command(*it))
```

Comment:
Rather than returning false on a bunch of the queue, we could maybe change this if-else to be 

```
if (!send_command(*it))
{
    break;
}
```
Otherwise we're wasting time with the mutex locked I think

Reply:
Hmm, I'm not sure. My thought of not breaking out of the loop is mainly that there can be commands 'single-framed' and others using transport protocols. Hence some might still succeed if others fail to send. Also since the mutex is solely around the CommandQueue, and in other places not used very often, I feel like the mutex being locked won't be really be a problem?

Reply:
Ah OK, I see, that makes sense